tests/orders/tests-edd-payment.php

```diff
@@ -316,14 +316,17 @@ public function test_payment_remove_fee() {
 		$this->assertEquals( 'Test Fee 2', $this->payment->fees[1]['label'] );
 
 		// Test that it saves to the DB
-		$payment_meta = edd_get_payment_meta( $this->payment->ID, '_edd_payment_meta', true );
-
-		$this->assertArrayHasKey( 'fees', $payment_meta );
-
-		$fees = $payment_meta['fees'];
+		$fees = edd_get_order_adjustments(
```

Comment:
I cannot replicate it by hand, but this is failing:

```php
$this->assertEquals( 3, count( $this->payment->fees ) );
```
The fee added with `$i = 0` is not being counted during the tests. However, copying the logic over and running it by hand, it is (I initially thought the issue was with doing an `empty` check, but the above code gives me `3` fees, just not with the tests).

Not sure how to resolve this one and would love any insight.

Reply:
@robincornett The issue is with this change (in various places):

```php
$id          = empty( $adjustment->type_key ) ? $adjustment->id : $adjustment->type_key;
$fees[ $id ] = $adjustment;
```

The reason is that, for `Test Fee 0`, the `type_key` is `0`, so it's hitting that `empty()` check, then falling back to using the ID instead, which is `1`, but then there's another fee that does have `type_key` `1`. So it's being overwritten.

There are two changes we should make here:

1. Instead of `empty( $adjustment->type_key )` we should use `is_null( $adjustment->type_key )`. This will allow `0` to be used, but not `null`.
2. After setting `$id` (but before assigning it) we should check if that already exists in the array, and if so, change it to something else. We could just append `_2` or something since the value isn't ultimately particularly important.

Full example:

```php
$id = is_null( $adjustment->type_key ) ? $adjustment->id : $adjustment->type_key;
if ( array_key_exists( $id, $fees ) ) {
	$id = $id . '_2';
}

$fees[ $id ] = $adjustment;
```